jaxrs-clients/src/main/java/com/palantir/remoting2/jaxrs/AbstractFeignJaxRsClientBuilder.java

```diff
@@ -0,0 +1,179 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.remoting2.jaxrs;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.net.HttpHeaders;
+import com.palantir.remoting2.clients.CipherSuites;
+import com.palantir.remoting2.clients.ClientBuilder;
+import com.palantir.remoting2.clients.ClientConfig;
+import com.palantir.remoting2.config.service.BasicCredentials;
+import com.palantir.remoting2.config.service.ProxyConfiguration;
+import com.palantir.remoting2.config.ssl.TrustContext;
+import com.palantir.remoting2.jaxrs.feignimpl.FailoverFeignTarget;
+import com.palantir.remoting2.jaxrs.feignimpl.FeignSerializableErrorErrorDecoder;
+import com.palantir.remoting2.jaxrs.feignimpl.GuavaOptionalAwareContract;
+import com.palantir.remoting2.jaxrs.feignimpl.Java8OptionalAwareContract;
+import com.palantir.remoting2.jaxrs.feignimpl.NeverRetryingBackoffStrategy;
+import com.palantir.remoting2.jaxrs.feignimpl.SlashEncodingContract;
+import com.palantir.remoting2.jaxrs.feignimpl.UserAgentInterceptor;
+import com.palantir.remoting2.tracing.okhttp3.OkhttpTraceInterceptor;
+import feign.CborDelegateDecoder;
+import feign.CborDelegateEncoder;
+import feign.Contract;
+import feign.Feign;
+import feign.GuavaOptionalAwareDecoder;
+import feign.InputStreamDelegateDecoder;
+import feign.InputStreamDelegateEncoder;
+import feign.Java8OptionalAwareDecoder;
+import feign.Logger;
+import feign.Request;
+import feign.TextDelegateDecoder;
+import feign.TextDelegateEncoder;
+import feign.codec.Decoder;
+import feign.jackson.JacksonDecoder;
+import feign.jackson.JacksonEncoder;
+import feign.jaxrs.JaxRsWithHeaderAndQueryMapContract;
+import feign.okhttp.OkHttpClient;
+import feign.slf4j.Slf4jLogger;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import okhttp3.ConnectionPool;
+import okhttp3.ConnectionSpec;
+import okhttp3.Credentials;
+import okhttp3.TlsVersion;
+
+abstract class AbstractFeignJaxRsClientBuilder extends ClientBuilder {
+
+    private final ClientConfig config;
+
+    AbstractFeignJaxRsClientBuilder(ClientConfig config) {
+        this.config = config;
+        Preconditions.checkArgument(config.maxNumRetries() == 0,
+                "Connection-level retries are not supported by %s", JaxRsClient.class.getSimpleName());
+    }
+
+    protected abstract ObjectMapper getObjectMapper();
```

Comment:
this unfortunately makes it very very easy for anyone outside this library to fuck with the object mapper. there's probably no way around this :( maybe service loading for adding the scala support at runtime? at the minimum, please add a class-level comment saying, that nobody outside this library must implement this class. thoughts, @markelliot ?